File: app/src/at-myhuynh/java/asiantech/internship/summer/java_core/ex25/ListPrime2.java
package asiantech.internship.summer.java_core.ex25;

import java.util.logging.Logger;

import asiantech.internship.summer.java_core.common.Common;

public class ListPrime2 {

    public static void main(String[] args) {
        Logger log = Logger.getLogger(ListPrime2.class.getName());
        for (int i = 1000000; i < 10000000; i++) {
            int total = Common.totalOfNumber(i);
            if (Common.isPrime(i) && Common.isChainOfSymmetry(String.valueOf(i)) && Common.isChainOfSymmetry(String.valueOf(total))) {
                log.info(String.valueOf(i));
            }
        }
    }

}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/ex25/ListPrime2.kt
package asiantech.internship.summer.kotlin.ex25

import asiantech.internship.summer.kotlin.common.isChainOfSymmetry
import asiantech.internship.summer.kotlin.common.isPrime
import asiantech.internship.summer.kotlin.common.totalOfNumber

object ListPrime2 {

    @JvmStatic
    fun main(args: Array<String>) {
        (1000000..9999999)
                .filter {
                    it.isPrime()
                            && it.toString().isChainOfSymmetry()
                            && it.totalOfNumber().toString().isChainOfSymmetry()
                }
                .forEach(::println)
    }
}
